api/test/service/upload.test.js

```diff
@@ -0,0 +1,367 @@
+    Array.from({ length: BATCH_MAX_SAFE_LIMIT + 1 }).map(() => randomCAR(40))
+  )
+  const root = cars[0].roots[0]
+  const shards = cars.map(c => c.cid)
+
+  /** @type {import('../../service/types').UploadItemOutput[]} */
+  const uploadAdd = await UploadCapabilities.add.invoke({
+    issuer: alice,
+    audience: uploadService,
+    with: spaceDid,
+    nb: { root, shards },
+    proofs: [proof]
+    // @ts-expect-error ʅʕ•ᴥ•ʔʃ
+  }).execute(connection)
+
+  // @ts-expect-error error is added by ucanto if it fails
+  t.not(uploadAdd.error, true, uploadAdd.message)
+  t.is(uploadAdd.length, shards.length)
+
+  // Validate DB before remove
+  const dbItemsBefore = await getSpaceItems(t.context.dynamoClient, spaceDid)
+  t.is(dbItemsBefore.length, shards.length)
+
+  // Remove Car from Space
+  await UploadCapabilities.remove.invoke({
+    issuer: alice,
+    audience: uploadService,
+    with: spaceDid,
+    nb: { root },
+    proofs: [proof]
+    // @ts-expect-error ʅʕ•ᴥ•ʔʃ
+  }).execute(connection)
+
+  // Validate DB after remove
+  const dbItemsAfter = await getSpaceItems(t.context.dynamoClient, spaceDid)
+  t.is(dbItemsAfter.length, 0)
+})
+
+test('store/list does not fail for empty list', async (t) => {
+  const uploadService = t.context.serviceDid
+  const alice = await Signer.generate()
+  const { proof, spaceDid } = await createSpace(alice)
+  const connection = await getClientConnection(uploadService, t.context)
+  
+  const uploadList = await UploadCapabilities.list.invoke({
+    issuer: alice,
+    audience: uploadService,
+    with: spaceDid,
+    proofs: [ proof ]
+    // @ts-expect-error ʅʕ•ᴥ•ʔʃ
+  }).execute(connection)
+
+  t.like(uploadList, { results: [], pageSize: 0 })
+})
+
+test('store/list returns entries previously uploaded by the user', async (t) => {
+  const uploadService = t.context.serviceDid
+  const alice = await Signer.generate()
+  const { proof, spaceDid } = await createSpace(alice)
+  const connection = await getClientConnection(uploadService, t.context)
+
+  // invoke multiple upload/add with proof
+  const cars = [
+    await randomCAR(128),
+    await randomCAR(128)
+  ]
+
+  for (const car of cars) {
+    await UploadCapabilities.add.invoke({
+      issuer: alice,
+      audience: uploadService,
+      with: spaceDid,
+      nb: { root: car.roots[0], shards: [car.cid] },
+      proofs: [proof]
+      // @ts-expect-error ʅʕ•ᴥ•ʔʃ
+    }).execute(connection)
+  }
+
+  /** @type {import('../../service/types').ListResponse<import('../../service/types').UploadItemOutput>} */
+  const uploadList = await UploadCapabilities.list.invoke({
+    issuer: alice,
+    audience: uploadService,
+    with: spaceDid,
+    proofs: [ proof ]
+    // @ts-expect-error ʅʕ•ᴥ•ʔʃ
+  }).execute(connection)
+
+  t.is(uploadList.pageSize, cars.length)
+
+  // Validate entries have given CARs
+  for (const entry of uploadList.results) {
+    t.truthy(cars.find(car => car.roots[0].toString() === entry.dataCID ))
+  }
```

Comment:
plz validate expected order of results list

Reply:
see https://github.com/web3-storage/upload-api/pull/26#discussion_r1026314813